scipy/interpolate/rbf.py

```diff
@@ -192,9 +193,14 @@ def __init__(self, *args, **kwargs):
         self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten()
                               for a in args[:-1]])
         self.N = self.xi.shape[-1]
-        self.di = np.asarray(args[-1]).flatten()
+        self.di = np.asarray(args[-1])
```

Comment:
This looks like a backwards compatibility break, in two ways:
- `di` is public, and you're changing its shape for >1-D input
- you're using `di.ndim` as an indication that the interpolation should now be multi-dimensional.

This should be avoided. I suspect that there's no better way than adding a new keyword argument (e.g. `flatten=True` or `mode='1-D') that preserves backwards compat but allows the user to enable multi-dim interpolation.